src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -1281,9 +1281,14 @@ def get_alias(self) -> Optional[ColumnAliasInfo]:
             return None
 
         alias_identifier_segment = next(
-            s for s in alias_expression_segment.segments if s.is_type("identifier")
+            (s for s in alias_expression_segment.segments if s.is_type("identifier")),
+            None,
         )
 
+        if alias_identifier_segment is None:
```

Comment:
Is this fix correct? In the test query, it looks like it *is* defining an alias name.

Reply:
I.e. are we fixing the bug or merely preventing it from crashing?

Reply:
It's not. It's a literal. See here: https://github.com/sqlfluff/sqlfluff/issues/2478#issuecomment-1021362087

Reply:
BTW this works:

```sql
SELECT @@IDENTITY AS ['Identity'];
```

Instead of this:

```sql
SELECT @@IDENTITY AS 'Identity';
```

Which would be a literal alias. But that's not valid T-SQL.

It's just a weird syntax. But think we are parsing it correctly. Not the first time we've seen that from T-SQL! 🤷‍♂️

Reply:
I understand that it's a literal (quoted) string. But isn't it creating and naming an alias? If not, what is the `AS` doing?

Reply:
Hmmm maybe it is supposed to be a literal alias. Digging...

Reply:
How often I wish for a test instance of these databases!

Reply:
This makes me wonder if the dialect is a bit wrong, I.e. defining aliases differently than other dialects. They can't all be identical, but making them as similar as practical is 💯.

Reply:
So single quotes are for string literals. Just confirmed you can't use them in BigQuery or Oracle and equivalent of above doesn't parse in SQLFluff for other dialects.

However T-SQL DOES support the above syntax. I've confirmed it.

I think adding a single_quoted identifiers to SQLFluff (and not confusing it with literals which are also single quotes) is a bigger task, and I'd rather merge this as is. I don't think it's "bad" to test the code before using it even if it turns out it's not needed for this T-SQL in future if we fix the single quote issue.

I'm keen to merge #2473 and this is currently a blocker.

WDYT @barrywhart ?

Reply:
Sounds good. Maybe create an issue for the T-SQL thing? Somebody can pick it up at some point.

Reply:
Actually I might have it and doesn't look too bad. So might add to this.

Decision then is whether to leave the current checks in place. I veering on yes as can't assume they will be set.

Reply:
I'm okay with leaving it. 👍

Reply:
Fix added if you want to have another look @barrywhart ?

Might need to add a `noqa` to my additional check as in theory shouldn't be called anymore. But we'll see when tests complete if coverage drops.